server/sonar-process-monitor/src/main/java/org/sonar/application/process/JavaCommandFactoryImpl.java

```diff
@@ -44,71 +44,82 @@
     "socksProxyHost",
     "socksProxyPort"};
 
+  private final AppSettings settings;
+
+  public JavaCommandFactoryImpl(AppSettings settings) {
+    this.settings = settings;
+  }
+
   @Override
-  public JavaCommand createESCommand(Props props, File workDir) {
-    return newJavaCommand(ProcessId.ELASTICSEARCH, props, workDir)
+  public JavaCommand createEsCommand() {
+    File homeDir = settings.getProps().nonNullValueAsFile(ProcessProperties.PATH_HOME);
+    return newJavaCommand(ProcessId.ELASTICSEARCH, homeDir)
       .addJavaOptions("-Djava.awt.headless=true")
-      .addJavaOptions(props.nonNullValue(ProcessProperties.SEARCH_JAVA_OPTS))
-      .addJavaOptions(props.nonNullValue(ProcessProperties.SEARCH_JAVA_ADDITIONAL_OPTS))
+      .addJavaOptions(settings.getProps().nonNullValue(ProcessProperties.SEARCH_JAVA_OPTS))
+      .addJavaOptions(settings.getProps().nonNullValue(ProcessProperties.SEARCH_JAVA_ADDITIONAL_OPTS))
       .setClassName("org.sonar.search.SearchServer")
       .addClasspath("./lib/common/*")
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "./lib/common/*" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1192)